src/Symfony/Component/Cache/CacheItem.php

```diff
@@ -100,6 +100,30 @@ public function expiresAfter($time)
     }
 
     /**
+     * Validates a cache key according to PSR-6.
+     *
+     * @param string $key The key to validate.
+     *
+     * @return string $key if it is valid.
+     *
+     * @throws InvalidArgumentException When $key is not valid.
+     */
+    public static function validateKey($key)
```

Comment:
I understand a validator as something that returns true/false depending whether the value is valid or not. Bu here we are throwing exceptions and returning the original value (potentially modified). To me it's very confusing.